mcp-server/main.py

```
@@ -281,50 +124,73 @@
             )
         logger.info("Token validation successful")
     except jwt.InvalidTokenError as e:
-        logger.error("Invalid token error: {}", str(e))
-        raise HTTPException(
-            status_code=401,
-            detail=f"Invalid token: {str(e)}"
-        )
+        logger.error(f"Invalid token error: {str(e)}")
+        raise ValueError(f"Invalid token: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)